src/components/Modals/TeleportModal.vue

```diff
@@ -43,7 +43,6 @@ import { defineComponent, ref } from 'vue';
 
 export default defineComponent({
   props: {
-    modelValue: { required: true, type: Boolean }, // true if the modal is visible, false otherwise
```

Comment:
So to confirm: Do we not need `modelValue` because we're using v-if?

Reply:
Yeah. Originally we had the v-if instead of TeleportModal itself, which is why we needed v-model, so that both TeleportModal and the parent of the modal had an updated display boolean. However, when we switched the v-if to be on the parent itself (for proper data resetting), there was no reason to use v-model or pass the boolean down through the children per @neketka. So this commit shouldn't functionally change anything, just replace the v-models with emits so the parents can set the display booleans true/false.